core/deployment/src/main/java/io/quarkus/deployment/ide/IDEDetectionProcessor.java

```diff
@@ -0,0 +1,116 @@
+        IDE_MARKER_FILES.forEach((file, ides) -> {
+            if (Files.exists(projectRoot.resolve(file))) {
+                result.addAll(ides);
+            }
+        });
+        return new IDEFileBuildItem(result);
+    }
+
+    @BuildStep
+    public IDERunningProcessBuildItem detectRunningIDEProcesses() {
+        Set<IDE> result = new HashSet<>(2);
+        List<ProcessInfo> processInfos = ProcessUtil.runningProcesses();
+        for (ProcessInfo processInfo : processInfos) {
+            for (Map.Entry<Predicate<ProcessInfo>, IDE> entry : IDE_PROCESSES.entrySet()) {
+                if (entry.getKey().test(processInfo)) {
+                    result.add(entry.getValue());
+                    break;
+                }
+            }
+        }
+        return new IDERunningProcessBuildItem(result);
+    }
+
+    // TODO: remove when we move to Java 11 and just call the methods of 'java.lang.ProcessHandle'
+    private static class ProcessUtil {
+
+        /**
+         * Returns a list of running processes
+         * Only works for Java 11+
+         */
+        public static List<ProcessInfo> runningProcesses() {
+            if (!JavaVersionUtil.isJava11OrHigher()) {
+                return Collections.emptyList();
+            }
+            // we can't use ProcessHandle directly as it is Java 9+, so we need to do reflection to the get the info we need
+            try {
+                Class processHandlerClass = Class.forName("java.lang.ProcessHandle");
+                Method allProcessesMethod = processHandlerClass.getMethod("allProcesses");
+                Method processHandleInfoMethod = processHandlerClass.getMethod("info");
+                Class processHandleInfoClass = Class.forName("java.lang.ProcessHandle$Info");
+                Method processHandleInfoCommandMethod = processHandleInfoClass.getMethod("command");
+                Stream<Object> allProcessesResult = (Stream<Object>) allProcessesMethod.invoke(null);
+                List<ProcessInfo> result = new ArrayList<>();
+                allProcessesResult.forEach(o -> {
+                    try {
+                        Object processHandleInfo = processHandleInfoMethod.invoke(o);
+                        Optional<String> command = (Optional<String>) processHandleInfoCommandMethod.invoke(processHandleInfo);
+                        if (command.isPresent()) {
+                            result.add(new ProcessInfo(command.get()));
+                        }
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+                return result;
+            } catch (Exception e) {
+                throw new RuntimeException("Unable to determine running processes", e);
+            }
+        }
+    }
+
+    private static class ProcessInfo {
+        // the executable pathname of the process.
+        private final String command;
```

Comment:
You can move this to the `IDE` enum you created above

Reply:
Things are really WIP at the moment, there is still a lot of moving around to do :)